core-bundle/src/Controller/ContentElement/AbstractContentElementController.php

```diff
@@ -59,5 +68,34 @@ protected function addSharedMaxAgeToResponse(Response $response, ContentModel $m
         $response->setSharedMaxAge(min($min));
     }
 
+    protected function isVisible(ContentModel $model, Request $request): bool
+    {
+        $isInvisible = $model->invisible || ($model->start && $model->start > time()) || ($model->stop && $model->stop <= time());
+
+        // The element is visible, so show it
+        if (!$isInvisible) {
+            return true;
+        }
+
+        return $this->isFrontendPreview($request);
+    }
+
+    protected function isFrontendPreview(Request $request): bool
+    {
+        $tokenChecker = $this->get('contao.security.token_checker');
+
+        // Preview mode is enabled, so show the element
+        if ($tokenChecker->hasBackendUser() && $tokenChecker->isPreviewMode()) {
```

Comment:
Does `isPreviewMode` not work without `hasBackendUser()` check first?

Reply:
I don't know. It used the same way in other places, for instance here: https://github.com/contao/contao/blob/2bbc4fd07bf29f8ad2c3f92853070f1b4d836a9b/core-bundle/src/Resources/contao/modules/ModuleArticle.php#L87